h2o-r/h2o-package/R/automl.R

```diff
@@ -329,7 +329,7 @@ h2o.getAutoML <- function(project_name) {
   automl_job <- .h2o.__remoteSend(h2oRestApiVersion = 99, method = "GET", page = paste0("AutoML/", project_name))
   leaderboard <- as.data.frame(automl_job["leaderboard_table"]$leaderboard_table)
   row.names(leaderboard) <- seq(nrow(leaderboard))
-  leaderboard <- as.h2o(leaderboard)
+  leaderboard <- as.h2o(leaderboard, paste(build_control$project_name, "leaderboard", sep="."))
   leaderboard[,2:length(leaderboard)] <- as.numeric(leaderboard[,2:length(leaderboard)])
```

Comment:
Is the idea to have a unique key for the leaderboard in an automl project? What is currently happening to cause this change?

Reply:
correct, that's the idea.
if you look at ticket and or unit test, after running 2 automls consecutively (2 different project names), the backend was throwing an exception when doing a Rapids operation on the 1st leaderboard, this occured because the second automl was overriding the leaderboard dataframe of the first one when this was converted by the R client from a table to a H2OFrame, leading to a corrupt object on client side (it didn't match it's backend representation).

Reply:
as mentioned in the rel-wright PR, I suspect this pattern `as.h2o(obj)` could lead to similar issues in different context as the key always turn out to be the variable name..., maybe we're just lucky enough for not using it too often., or maybe it's not always raising exception but just manipulating wrong object

Reply:
FYI: for some reason, can't see your comments on rel-wright PR, but received emails: looks like GitHub is having some issues...

Reply:
ideally, should be fixed directly in all those `as.h2o` methods.
this pattern is basically the culprit:
```
  if( destination_frame=="" )
    destination_frame <- deparse(substitute(x))
```
`as.h2o.H2OFrame` seems to do it better:
```
 if( destination_frame=="" ) {
    subx <- destination_frame.guess(deparse(substitute(x)))
    destination_frame <- .key.make(if(nzchar(subx)) subx else "copy")
  }
```
wdyt?